Automattic/wp-calypso
# Adjust commercial page styling for WPCOM

Related to #80888

## Proposed Changes

* Introduce `getIsSiteWPCOM` to adjust styling for the Stats Commercial purchase page of WPCOM sites. The context is https://github.com/Automattic/wp-calypso/pull/81359#pullrequestreview-1609984026.
* Gate the purchase redirection behind the feature flag `stats/paid-wpcom-stats`.

## Testing Instructions



* Spin up the change with the Calypso Live link.
* Navigate to the Stats Commercial purchase page of a `Jetpack` site with the feature flag: `/stats/purchase/{jetpack-site-slug}?flags=stats/paid-wpcom-stats,stats/type-detection&productType=commercial`.
* Ensure the page styling works as previously.
* Navigate to the Stats Commercial purchase page of a `Simple or Atomic` site with the feature flag: `/stats/purchase/{wpcom-site-slug}?flags=stats/paid-wpcom-stats,stats/type-detection&productType=commercial`.
* Ensure the page styling is in line with the design.

<img width="1097" alt="截圖 2023-09-05 上午12 34 31" src="https://github.com/Automattic/wp-calypso/assets/6869813/26ebd336-ea50-4206-8794-b543a5ca85d4">

|Jetpack site|WPCOM site|
|-|-|
|<img width="1104" alt="截圖 2023-09-05 上午12 20 15" src="https://github.com/Automattic/wp-calypso/assets/6869813/944dd82a-fa12-4055-8966-588dc191f2e0">|<img width="1095" alt="截圖 2023-09-06 上午12 06 33" src="https://github.com/Automattic/wp-calypso/assets/6869813/02c8ba0c-f3c5-4295-8997-b67125ff0bbc">|

## Pre-merge Checklist
